resolver/src/main/java/io/netty/resolver/InetNameResolver.java

```diff
@@ -40,32 +37,15 @@
      */
     protected InetNameResolver(EventExecutor executor) {
         super(executor);
-        loopbackAddress = SocketUtils.loopbackAddress();
     }
 
     /**
+     * @deprecated Use {@link SocketUtils#loopbackAddress()}.
      * Returns the {@link InetAddress} for loopback.
      */
+    @Deprecated
     protected InetAddress loopbackAddress() {
-        return loopbackAddress;
-    }
-
-    @Override
-    public Future<InetAddress> resolve(String inetHost, Promise<InetAddress> promise) {
-        if (inetHost == null || inetHost.isEmpty()) {
-            // If an empty hostname is used we should use "localhost", just like InetAddress.getByName(...) does.
-            return promise.setSuccess(loopbackAddress());
-        }
-        return super.resolve(inetHost, promise);
-    }
-
-    @Override
-    public Future<List<InetAddress>> resolveAll(String inetHost, Promise<List<InetAddress>> promise) {
-        if (inetHost == null || inetHost.isEmpty()) {
-            // If an empty hostname is used we should use "localhost", just like InetAddress.getByName(...) does.
-            return promise.setSuccess(Collections.singletonList(loopbackAddress()));
-        }
-        return super.resolveAll(inetHost, promise);
+        return SocketUtils.loopbackAddress();
```

Comment:
I originally wanted to remove this method but wasn't sure if this would be OK (see https://github.com/netty/netty/pull/6331#discussion_r100219110). I don't think it should be used but if we can't remove the method and you think it may be used then I can bring the caching back.